hawc/apps/study/api.py

```diff
@@ -47,6 +55,27 @@ def types(self, request):
         study_types = self.model.STUDY_TYPE_FIELDS
         return Response(study_types)
 
+    def create(self, request):
+
+        data = request.data.dict() if isinstance(request.data, QueryDict) else request.data
+        reference_id = tryParseInt(data.pop("reference_id", -1), -1)
+
+        try:
+            reference = Reference.objects.get(id=reference_id)
+        except ObjectDoesNotExist:
+            raise ValidationError(f"Reference ID does not exist.")
+        data["assessment"] = reference.assessment.id
+
```

Comment:
I handled the above this way so that I could check permissions of the reference object. I suppose I could pass the request to the serializer as context and test there, but I'm not sure if thats appropriate for the serailizer. Let me know!

Reply:
I can see why you put it here, but it basically means you have to entirely rewrite the create method.   It'd probably be better to put everything in the serializer, and then if you want you can add this:

```python
def create(self, request):
    # permissions check not here; see serializer validation
    return super().create(request)
```

Reply:
Done!